framework/src/modules/nft/method.ts

```diff
@@ -854,4 +859,125 @@ export class NFTMethod extends BaseMethod {
 			collectionID,
 		});
 	}
+
+	public async recover(
+		methodContext: MethodContext,
+		terminatedChainID: Buffer,
+		substorePrefix: Buffer,
+		storeKey: Buffer,
+		storeValue: Buffer,
+	): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+		const nftID = storeKey;
+		let isDecodable = true;
+		let decodedValue: NFTStoreData;
+		try {
+			decodedValue = codec.decode<NFTStoreData>(nftStoreSchema, storeValue);
+		} catch (error) {
+			isDecodable = false;
+		}
+
+		if (
+			!substorePrefix.equals(nftStore.subStorePrefix) ||
+			storeKey.length !== LENGTH_NFT_ID ||
+			!isDecodable
+		) {
+			this.events.get(RecoverEvent).error(
+				methodContext,
+				{
+					terminatedChainID,
+					nftID,
+				},
+				NftEventResult.RESULT_RECOVER_FAIL_INVALID_INPUTS,
+			);
+			throw new Error('Invalid inputs');
+		}
+
+		const nftChainID = this.getChainID(nftID);
+		const ownChainID = this._internalMethod.getOwnChainID();
+		if (!nftChainID.equals(ownChainID)) {
+			this.events.get(RecoverEvent).error(
+				methodContext,
+				{
+					terminatedChainID,
+					nftID,
+				},
+				NftEventResult.RESULT_INITIATED_BY_NONNATIVE_CHAIN,
+			);
+			throw new Error('Recovery called by a foreign chain');
+		}
+
+		const nftData = await nftStore.get(methodContext, nftID);
+		if (!nftData.owner.equals(terminatedChainID)) {
+			this.events.get(RecoverEvent).error(
+				methodContext,
+				{
+					terminatedChainID,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_NOT_ESCROWED,
+			);
+			throw new Error('NFT was not escrowed to terminated chain');
+		}
+
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const storeValueOwner = decodedValue!.owner;
+		if (storeValueOwner.length !== LENGTH_ADDRESS) {
+			this.events.get(RecoverEvent).error(
+				methodContext,
+				{
+					terminatedChainID,
+					nftID,
+				},
+				NftEventResult.RESULT_INVALID_ACCOUNT,
+			);
+			throw new Error('Invalid account information');
+		}
+
+		const escrowStore = this.stores.get(EscrowStore);
+		nftData.owner = storeValueOwner;
```

Comment:
I believe we agreed on implementing [this part](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md?plain=1#L1939-L1941) of the LIP pseudocode here: 
```
storedAttributes = NFTStore[nftID].attributesArray
receivedAttributes = nftValue.attributes
NFTStore[nftID].attributes = getNewAttributes(nftID, storedAttributes, receivedAttributes)
```

Reply:
maybe I misunderstood the previous discussion as - we want to only indicate via comment that `getNewAttributes` can be used here and not actually use it now (as it'd be a redundant call). I realize it could be confusing or not clear enough so I added the additional(redundant) logic to strictly match the LIP as you suggest. Same for `CrossChainTransferCommand` execute in `cc_transfer.ts`

Reply:
I would say it still needs to be done. As discussed earlier, the idea is not only to define `getNewAttributes`, but also show to the developers in what context it is to be used.